common/browsered/field-validation.js

```diff
@@ -95,23 +98,42 @@ function applyErrorMessaging (form, field, result) {
   let formGroup = field.closest('.form-group')
   if (!formGroup.classList.contains('error')) {
     formGroup.classList.add('error')
-    let label = document.querySelector('label[for="' + field.name + '"]')
-    let errorLabel = label.getAttribute('data-error-label')
-    if (errorLabel) {
-      result = errorLabel
+    const errorLegendElement = formGroup.querySelector('legend')
+    if (errorLegendElement === null) {
+      const errorElement = document.querySelector('label[for="' + field.name + '"]')
```

Comment:
This if/else has a lot of duplication. Could make simpler

Reply:
Thanks, not sure how to refactor this, If you can help with some example :)
Basically, the only duplicated code is:
```
      if (errorLabel) {
        result = errorLabel
      }
```
we have different `insertAdjacentHTML` and `appendChild` and also use different `querySelector`, could you provide maybe an example? Thanks

Reply:
We are going to remove the error messages with the refactoring of https://github.com/alphagov/pay-direct-debit-frontend/pull/40#discussion_r165375003 and the logic should be simpler